graphql/resolvers.go

```diff
@@ -189,6 +189,40 @@ func (r *volumeResolver) Host(ctx context.Context, obj *restModel.APIVolume) (*r
 	return &apiHost, nil
 }
 
+func (r *queryResolver) HasVersion(ctx context.Context, id string) (bool, error) {
+	// We do not check the error here because we only want to return an error if the id is not
+	// a valid patch or version id.
+	v, err := r.sc.FindVersionById(id)
+	if err != nil {
+		apiErr, ok := err.(gimlet.ErrorResponse)
+		if !ok {
+			return false, InternalServerError.Send(ctx, fmt.Sprintf("Error finding version %s: %s", id, err.Error()))
+		}
+		if apiErr.StatusCode != http.StatusNotFound {
+			return false, InternalServerError.Send(ctx, fmt.Sprintf("Error finding version %s: %s", id, err.Error()))
+		}
+	}
+	if v != nil {
+		return true, nil
+	}
+	if patch.IsValidId(id) {
+		p, err := patch.FindOneId(id)
+		if err != nil {
+			apiErr, ok := err.(gimlet.ErrorResponse)
+			if !ok {
+				return false, InternalServerError.Send(ctx, fmt.Sprintf("Error finding version %s: %s", id, err.Error()))
+			}
+			if apiErr.StatusCode != http.StatusNotFound {
```

Comment:
Don't need to check status code here because you aren't using a connector.
Also actually sorry I didn't realize before; if you don't want to do the StatusNotFound drama above, you can use the actual DB function get the version

Reply:
Thanks the drama has been dropped!